contributors/design-proposals/lifecycle-hooks.md

```diff
@@ -0,0 +1,157 @@
+# Lifecycle Hooks
+
+**Author**: @tnozicka
+
+**Status**: Proposal
+
+**RFC Issue**: https://github.com/kubernetes/kubernetes/issues/14512
+
+## Abstract
+The intent of this proposal it to support lifecycle hooks in all objects having rolling/recreate update; currently those are Deployments, StatefulSets and DaemonSets. Lifecycle hooks should allow users to better customize updates (rolling/recreate) without having to write their own controllers. 
```

Comment:
Can you please split these into separate lines? Otherwise, it's difficult to review.

Reply:
done